windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -2059,5 +2152,132 @@ function GetKeymanWebCompiledFileName(const FileName: WideString): WideString;
   Result := GetKeymanWebCompiledFileName(FileName, GetKeyboardVersionString(FileName));
 end;
 
+///
+/// If debug mode, then returns Javascript code necessary for
+/// accessing constants in the compiled keyboard
+///
+/// @return string of JavaScript code
+///
+function TCompileKeymanWeb.JavaScript_SetupDebug: string;
+begin
+  if FDebug then
+    Result := 'var modCodes = tavultesoft.keymanweb.osk.modifierCodes;'+nl+
+              FTabStop+'var keyCodes = tavultesoft.keymanweb.osk.keyCodes;'+nl
+  else
+    Result := '';
+end;
+
+///
+/// Converts a modifier bit mask integer into its component bit flags
+///
+/// @param FBitMask A KMX modifier bitmask value
+///
+/// @return string of JavaScript code, e.g. 'modCodes.SHIFT | modCodes.CTRL /* 0x0030 */'
+///
+function TCompileKeymanWeb.FormatModifierAsBitflags(FBitMask: Cardinal): string;
+const
+  mask: array[0..14] of string = (
+    'LCTRL',             // 0X0001
+    'RCTRL',             // 0X0002
+    'LALT',              // 0X0004
+    'RALT',              // 0X0008
+
+    'SHIFT',             // 0X0010
+    'CTRL',              // 0X0020
+    'ALT',               // 0X0040
+
+    '???',               // Reserved
```

Comment:
If these are the `osk.modifierCodes` from #300, will we need to add

`"???":0x0080` for Reserved?

Reply:
No, this value will never be generated; we just need to have a placeholder in the array so it lines up with the bitmasks.